driver-core/src/main/com/mongodb/client/model/expressions/Expressions.java

```diff
@@ -228,6 +286,18 @@ public static DocumentExpression of(final Bson document) {
                 document.toBsonDocument(BsonDocument.class, cr))));
     }
 
+    /**
+     * The null value in the context of the MongoDB Query Language (MQL).
+     *
+     * <p>The null value is not part of, and cannot be used as if it were part of,
+     * any other type. It has no explicit type of its own. Instead of checking for
+     * null, users should check for their expected type, via methods such as
+     * {@link Expression#isNumberOr(NumberExpression)}. Where the null value must
+     * be checked explicitly, users may use {@link Branches#isNull} within
+     * {@link Expression#switchOn}.
```

Comment:
_"Instead of checking for null, users should check for their expected type"_ - as a user I don't understand why I should check for other types, which null is not of, when I need to check for null (as someone who knows how `isNumberOr` works, I understand what you wanted to convey, but I don't think it was successfully conveyed)

Reply:
I'll think about this, but I am unsure what else to say.

The idea, just to make it more clear, is that there is no null check, because there is no null type to convert to and then use "as a null". So the only time you should need to null-check is in the context of a number of switch branches. This is not usually the case in Java, because an `Integer` variable can equal null.